backend/app.py

```diff
@@ -1608,6 +1608,170 @@ def financial_assistant(subscriptionId):
+
+        return create_success_response(
+            {
+                "message": "Financial Assistant removed from subscription successfully.",
+                "subscription": {
+                    "id": updated_subscription.id,
+                    "status": updated_subscription.status,
+                    "current_period_end": updated_subscription.current_period_end,
+                },
+            }
+        )
+
+    except stripe.error.InvalidRequestError as e:
+        logging.error(f"Stripe API error: {str(e)}")
+        return create_error_response(
+            "Invalid Subscription ID", HTTPStatus.NOT_FOUND
+        )
+    except stripe.error.StripeError as e:
+        logging.error(f"Stripe API error: {str(e)}")
+        return create_error_response(
+            "An error occurred while processing your request", HTTPStatus.BAD_REQUEST
+        )
+    except NotFound as e:
+        logging.warning(f"Not found: {str(e)}")
+        return create_error_response(str(e), HTTPStatus.NOT_FOUND)
+    except Exception as e:
+        logging.exception(f"Unexpected error: {str(e)}")
+        return create_error_response(
+            "An unexpected error occurred", HTTPStatus.INTERNAL_SERVER_ERROR
+        )
+
+#CHECK STATUS SUBSCRIPTION FA (FINANCIAL ASSITANT)
+@app.route("/api/subscription/<subscriptionId>/financialAssistant/status", methods=["GET"])
+@require_client_principal  # Security: Enforce authentication
+def get_financial_assistant_status(subscriptionId):
+    """
+    Check if Financial Assistant is added to a subscription.
+
+    Args:
+        subscriptionId (str): Unique Stripe Subscription ID
+
+    Returns:
+        JsonResponse: Response indicating if Financial Assistant is active in the subscription.
+        Success format:
+        {
+            "data": {
+                "financial_assistant_active": true,
+                "subscription": {
+                    "id": "<subscriptionId>",
+                    "status": "active"
+                }
+            }
+        }
+        
+    Raises:
+        NotFound: If the subscription is not found. HttpCode: 404
+        Unauthorized: If client principal ID is missing. HttpCode: 401
+    """
+    try:
+        # Consultar la suscripción en Stripe
+        subscription = stripe.Subscription.retrieve(subscriptionId)
+        
+        # Revisar si el Financial Assistant está activo en la suscripción
+        financial_assistant_active = any(
+            item.price.id == FINANCIAL_ASSISTANT_PRICE_ID for item in subscription["items"]["data"]
+        )
+        
+        # Responder con el estado del Financial Assistant
+        return jsonify({
+            "data": {
+                "financial_assistant_active": financial_assistant_active,
+                "subscription": {
+                    "id": subscription.id,
+                    "status": subscription.status
+                }
+            }
+        }), HTTPStatus.OK
+
```

Comment:
@ramirezmorac2 at this point you might want to return 404 or raise and 404 exception because you didn't found a  financial assistant

Reply:
@ramirezmorac2 you might want to implement something like this instead:

```python
        # Find the Financial Assistant item in the subscription
        financial_assistant_item = next(
            (item for item in subscription["items"]["data"] 
             if item.price.id == FINANCIAL_ASSISTANT_PRICE_ID),
            None
        )
        
        # Return 404 if Financial Assistant is not found in subscription
        if not financial_assistant_item:
            logging.info(f"Financial Assistant not found in subscription: {subscriptionId}")
            return jsonify({
                "error": {
                    "message": "Financial Assistant not found in this subscription",
                    "status": 404
                }
            }), HTTPStatus.NOT_FOUND
            
        # Return Financial Assistant data when found
        return jsonify({
            "data": {
                "id": subscription.id,
                "subscription_status": subscription.status,
                "price_id": financial_assistant_item.price.id,
                "product_name": financial_assistant_item.price.product.name,
                "created": financial_assistant_item.created,
                "current_period_start": subscription.current_period_start,
                "current_period_end": subscription.current_period_end
            }
        }), HTTPStatus.OK
```